uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-request-popup-details-tab.component.js

```diff
@@ -464,7 +481,12 @@ define([
      * Calculates and updates opening and closing balances
      */
     function _calculateOpeningAndClosingBalance () {
-      vm.balance.opening = vm.selectedAbsenceType.remainder;
+      if (originalOpeningBalance &&
+        originalOpeningBalance.absenceTypeId === vm.selectedAbsenceType.id) {
```

Comment:
Please leave a comment in the code, explaining why checking the type is necessary.